lib/frizz/ignorance.rb

```diff
@@ -2,13 +2,30 @@ module Frizz
   class Ignorance
     attr_reader :patterns
 
-    def initialize(patterns)
+    def initialize(patterns, prefer_gzip)
       @patterns = patterns || []
+      @prefer_gzip = prefer_gzip
     end
 
-    def ignore?(path)
+    def ignore?(local_path, full_path)
+      ignore_matched_pattern?(local_path) || ignore_for_gzip_version?(full_path)
+    end
+
+    private
+
+    def ignore_matched_pattern?(local_path)
       return false unless patterns.count
-      patterns.any? { |p| ::File.fnmatch(p, path) }
+
+      patterns.any? { |p| ::File.fnmatch(p, local_path) }
+    end
+
+    def ignore_for_gzip_version?(full_path)
+      return false unless @prefer_gzip
```

Comment:
See `attr_reader` on line 3.. add `prefer_gzip` to that and the ivar will get an attr reader method `prefer_gzip`. I prefer this approach over directly accessing ivars as it makes for a simple path to refactor ivars to be backed by more complex methods